docs-src/content/sessions/actions.md

```diff
@@ -110,14 +110,38 @@ func processActions(sessionOutput: Cobaltspeech_Diatheke_SessionOutput) {
 			self.handleReply(replyAction)
 		case .command(let commandAction):
 			// The CommandAction will involve a session update
-			self.handleCommand(commandAction
+			self.handleCommand(commandAction)
 		}
 	}
 }
 {{< /tab >}}
 
 {{< tab "Java/Android" "java" >}}
-// Example coming soon!
+private void executeActions(SessionOutput resp) {
+    List<ActionData> actionsToExecute = resp.getActionListList();
+
+    for (ActionData action : actions) {
+        switch (action.getActionCase()) {
+            case INPUT:
+                WaitForUserAction input = action.getInput();
+                break;
+
+            case COMMAND:
+                CommandAction cmd = action.getCommand();
+                handleCommandRequest(cmd);
```

Comment:
Isn't this setup going to cause problems with recursion? Someone will call `executeActions` first, and then `handleCommandRequest` and `waitForInput` (which should probably be added to the `INPUT` case above), end by calling `executeActions`. So the function stack will look like this:

```
executeActions (called from handleCommandRequest)
handleCommandRequest
executeActions (first call)
```

Reply:
Wait, is the Swift example doing this too?

Reply:
I don't think that this should cause any problems with recursion.  I also asked that topic on the Swift PR, and @kapitoshka438 said he couldn't get it to cause any problems.
With that said, I could spin up a new thread each time we talk to the server, but I'm not sure if the tradeoff is worth it (for an example in the documentation)